test/parallel/test-process-warnings.mjs

```diff
@@ -0,0 +1,162 @@
+      assert.strictEqual(signal, null);
+    });
+
+    it('should silence deprecation warnings DEP1 and DEP2', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        '--disable-warning=DEP1',
+        '--disable-warning=DEP2',
+        fixturePath,
+      ]);
+
+      assert.strictEqual(stdout, '');
+      assert.doesNotMatch(stderr, dep1Message);
+      assert.doesNotMatch(stderr, dep2Message);
+      assert.match(stderr, experimentalWarningMessage);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should silence all deprecation warnings using type DeprecationWarning', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        '--disable-warning=DeprecationWarning',
+        fixturePath,
+      ]);
+
+      assert.strictEqual(stdout, '');
+      assert.doesNotMatch(stderr, dep1Message);
+      assert.doesNotMatch(stderr, dep2Message);
+      assert.match(stderr, experimentalWarningMessage);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should silence all experimental warnings using type ExperimentalWarning', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        '--disable-warning=ExperimentalWarning',
+        fixturePath,
+      ]);
+
+      assert.strictEqual(stdout, '');
+      assert.match(stderr, dep1Message);
+      assert.match(stderr, dep2Message);
+      assert.doesNotMatch(stderr, experimentalWarningMessage);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should pass down option to worker', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        '--disable-warning=DEP2',
+        fixturePathWorker,
+      ]);
+
+      assert.strictEqual(stdout, '');
+      assert.match(stderr, dep1Message);
+      assert.doesNotMatch(stderr, dep2Message);
+      assert.match(stderr, experimentalWarningMessage);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should not support a comma separated list', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        '--disable-warning=DEP1,DEP2',
+        fixturePathWorker,
+      ]);
+
+      assert.strictEqual(stdout, '');
+      assert.match(stderr, dep1Message);
+      assert.match(stderr, dep2Message);
+      assert.match(stderr, experimentalWarningMessage);
+      assert.strictEqual(code, 0);
+      assert.strictEqual(signal, null);
+    });
+
+    it('should be specifiable in NODE_OPTIONS', async () => {
+      const { stdout, stderr, code, signal } = await spawnPromisified(process.execPath, [
+        fixturePath,
+      ], {
+        env: {
+          NODE_OPTIONS: '--disable-warning=DEP2'
+        }
```

Comment:
Failures in https://ci.nodejs.org/job/node-test-commit-linux-containered/40487/ are due to environment variables (e.g. `LD_LIBRARY_PATH`) that are required to run Node.js in those configurations being lost when launching the child process.

Reply:
so I have to forward them along?

Reply:
Yes. For tests, normally we extend the `env` for child processes rather than completely replacing it.

Reply:
Sgtm will fix shortly and rerun CI